lib/web_ui/test/canvaskit/embedded_views_test.dart

```diff
@@ -0,0 +1,64 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+// @dart = 2.12
+import 'dart:async';
+import 'dart:html' as html;
+
+import 'package:ui/src/engine.dart';
+import 'package:ui/ui.dart' as ui;
+
+import 'package:test/bootstrap/browser.dart';
+import 'package:test/test.dart';
+
+import 'common.dart';
+
+const MethodCodec codec = StandardMethodCodec();
+final EngineSingletonFlutterWindow window = EngineSingletonFlutterWindow(0, EnginePlatformDispatcher.instance);
+
+void main() {
+  internalBootstrapBrowserTest(() => testMain);
+}
+
+void testMain() {
+  group('HtmlViewEmbedder', () {
+    setUpCanvasKitTest();
+
+    test('embeds interactive platform views', () async {
+      ui.platformViewRegistry.registerViewFactory(
+        'test-platform-view',
```

Comment:
This worries me a bit because there's no way to clean it up. Tests may conflict with each other since factory names are global.